### PR TITLE
Add the ability to snap auto sampling to specific intervals per datasource.

### PR DESCRIPTION
**Pull request by ddimensia**

### CODE REVIEW

**Comment by ddimensia** [comment]
Screen Shot 2019-05-06 at 1 01 02 PM

File: dist/utils/time_unit_utils.ts [hunk 1]
```diff
@@ -23,11 +27,81 @@ export class TimeUnitUtils {
         return TimeUnit[unit];
     }
 
+    public static ceilingToAvailableUnit(interval: string, availableUnits: UnitValue[]): [string, string] {
+      const intervalMillis = this.intervalToMillis(interval);
+
+      // match the first available unit, available Units must be ordered
```
Comment (vjkoskela) [inline dist/utils/time_unit_utils.ts]: Is something validating or sorting these when the configuration is specified?

File: specs/core/request/parameter_object_builder.test.ts [hunk 2]
```diff
@@ -48,4 +56,17 @@ describe("ParameterObjectBuilder", () => {
         // then
         parameterObject.sampling.value.should.equal(INTERVAL_VALUE);
     });
+
+    it("should use interval values snapped to the closes value when auto value is enabled", () => {
```
Comment (vjkoskela) [inline specs/core/request/parameter_object_builder.test.ts]: type-o: `closest`

File: dist/utils/time_unit_utils.ts (hunk 1 above)
Comment (ddimensia) [reply dist/utils/time_unit_utils.ts]: The code sorts the list after it converts it.  There is handling for bad data where it eliminates values that don't map to a unit/value.  I added tests to validate that behavior.

File: specs/core/request/parameter_object_builder.test.ts (hunk 2 above)
Comment (ddimensia) [reply specs/core/request/parameter_object_builder.test.ts]: fixed

**Pull request merged**